apps/runtime/internal/cacheintercept/cacheintercept_test.go

```
@@ -0,0 +1,211 @@
+		t.Fatal("a client that does not trust the ephemeral CA accepted the leaf")
+	}
+}
+
+// New must not keep the caller's URL: a later mutation of it would move the
+// forward target past the scheme and host validation.
+func TestUpstreamMutationAfterNewIsIgnored(t *testing.T) {
+	w := setup(t, http.StatusOK, "ok")
+	w.upstreamURL.Scheme = "http"
+	w.upstreamURL.Host = "wrong.invalid:1"
+
+	resp, err := w.client.Get("https://" + cacheca.CacheHost + "/")
```

golangci-lint (2.12.2)

[error] (*net/http.Client).Get must not be called. use (*net/http.Client).Do(*http.Request)

(noctx)